libs/api/domains/auth/src/lib/loaders/client.loader.ts

```diff
@@ -1,29 +1,62 @@
-import { Injectable } from '@nestjs/common'
+import {
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common'
 import DataLoader from 'dataloader'
 
+import { GraphQLContext, User } from '@island.is/auth-nest-tools'
 import { NestDataLoader } from '@island.is/nest/dataloader'
 
 import { ClientInput } from '../dto/client.input'
 import { Client } from '../models/client.model'
+import { ClientsService } from '../services/clients.service'
+import { DomainService } from '../services/domain.service'
 
 export type ClientDataLoader = DataLoader<ClientInput, Client>
 
 @Injectable()
 export class ClientLoader implements NestDataLoader<ClientInput, Client> {
+  constructor(
+    private readonly clientService: ClientsService,
+    private readonly domainsService: DomainService,
+  ) {}
+
   keyFn(input: ClientInput): string {
     return `${input.lang}##${input.clientId}`
   }
 
-  async loadClients(inputs: readonly ClientInput[]): Promise<Array<Client>> {
-    // Todo: Add call to client endpoint in delegation api when ready.
+  async loadClients(
+    user: User | undefined,
+    inputs: readonly ClientInput[],
+  ): Promise<Array<Client | Error>> {
+    if (!user) {
+      throw new UnauthorizedException()
+    }
+
+    // Only support one language at a time.
+    const lang = inputs[0].lang
+    const authClients = await this.clientService.getClients(user, {
+      lang,
+      clientIds: inputs.map((input) => input.clientId),
+    })
+    const domains = await this.domainsService.getDomains(user, { lang })
```

Comment:
~Could use a field resolver and share the domain dataloader for this. Saves a request in case the query isn't requesting domain fields.